newIDE/app/src/InstancesEditor/InstancesResizer.js

```diff
@@ -1,15 +1,56 @@
 // @flow
-type Dimension = {|
-  width: number,
-  height: number,
-|};
+import Rectangle from '../Utils/Rectangle';
+import { roundPositionForResizing } from '../Utils/GridHelpers';
+
+export type ResizeGrabbingLocation =
+  | 'TopLeft'
+  | 'BottomLeft'
+  | 'BottomRight'
+  | 'TopRight'
+  | 'Top'
+  | 'Left'
+  | 'Bottom'
+  | 'Right';
+
+export const resizeGrabbingLocationValues = [
+  'TopLeft',
+  'BottomLeft',
+  'BottomRight',
+  'TopRight',
+  'Top',
+  'Left',
+  'Bottom',
+  'Right',
+];
+
+export const resizeGrabbingRelativePositions = {
+  TopLeft: [0, 0],
+  BottomLeft: [0, 1],
+  BottomRight: [1, 1],
+  TopRight: [1, 0],
+  Top: [0.5, 0],
+  Left: [0, 0.5],
+  Bottom: [0.5, 1],
+  Right: [1, 0.5],
+};
+
+export const canMoveOnX = (location: ResizeGrabbingLocation) =>
+  location !== 'Top' && location !== 'Bottom';
+
+export const canMoveOnY = (location: ResizeGrabbingLocation) =>
+  location !== 'Left' && location !== 'Right';
 
 export default class InstancesResizer {
   instanceMeasurer: any;
   options: Object;
-  instanceSizes: { [number]: Dimension };
-  totalDeltaX: number;
-  totalDeltaY: number;
+  _initialSelectionAABB: ?Rectangle = null;
+  _unrotatedInstanceAABBs: { [number]: Rectangle } = {};
+  _instanceAABBs: { [number]: Rectangle } = {};
+  _instancePositions: { [number]: { x: number, y: number } } = {};
+  totalDeltaX: number = 0;
```

Comment:
Let's add a comment for these: "The coordinates of the vector of the resize being done", as the class is becoming more complex.